test/passes/issue2788.txt

```diff
@@ -0,0 +1,684 @@
+ (type $none_=>_v128 (func (result v128)))
+ (type $i32_=>_v128 (func (param i32) (result v128)))
+ (type $f32_i64_funcref_f64_f64_funcref_=>_v128 (func (param f32 i64 funcref f64 f64 funcref) (result v128)))
+ (type $v128_=>_v128 (func (param v128) (result v128)))
+ (type $nullref_v128_i64_=>_v128 (func (param nullref v128 i64) (result v128)))
+ (type $f32_funcref_=>_v128_i64 (func (param f32 funcref) (result v128 i64)))
+ (type $nullref_=>_v128_i64_funcref_anyref_nullref (func (param nullref) (result v128 i64 funcref anyref nullref)))
+ (type $none_=>_v128_f32_v128_funcref_i64 (func (result v128 f32 v128 funcref i64)))
+ (type $anyref_=>_v128_anyref_i64 (func (param anyref) (result v128 anyref i64)))
+ (type $none_=>_v128_nullref_i64 (func (result v128 nullref i64)))
+ (type $anyref_nullref_=>_funcref (func (param anyref nullref) (result funcref)))
+ (type $v128_f32_=>_funcref_i32_f32_f32_v128 (func (param v128 f32) (result funcref i32 f32 f32 v128)))
+ (type $i32_=>_anyref (func (param i32) (result anyref)))
+ (type $i64_=>_anyref (func (param i64) (result anyref)))
+ (type $f32_=>_anyref (func (param f32) (result anyref)))
+ (type $f32_f64_anyref_anyref_=>_anyref (func (param f32 f64 anyref anyref) (result anyref)))
+ (type $v128_f64_i32_i64_anyref_nullref_=>_anyref (func (param v128 f64 i32 i64 anyref nullref) (result anyref)))
+ (type $funcref_f64_v128_f32_funcref_anyref_=>_anyref (func (param funcref f64 v128 f32 funcref anyref) (result anyref)))
+ (type $nullref_=>_anyref (func (param nullref) (result anyref)))
+ (type $f32_=>_anyref_funcref (func (param f32) (result anyref funcref)))
+ (type $i64_funcref_=>_anyref_anyref_i32 (func (param i64 funcref) (result anyref anyref i32)))
+ (type $anyref_f32_=>_anyref_anyref_nullref_anyref_nullref (func (param anyref f32) (result anyref anyref nullref anyref nullref)))
+ (type $i32_=>_anyref_nullref_nullref_f32 (func (param i32) (result anyref nullref nullref f32)))
+ (type $none_=>_nullref (func (result nullref)))
+ (type $i64_=>_nullref (func (param i64) (result nullref)))
+ (type $i64_anyref_nullref_=>_nullref (func (param i64 anyref nullref) (result nullref)))
+ (type $f64_v128_f32_=>_nullref (func (param f64 v128 f32) (result nullref)))
+ (type $v128_funcref_funcref_f32_=>_nullref (func (param v128 funcref funcref f32) (result nullref)))
+ (type $nullref_f32_f64_i64_v128_v128_=>_nullref_f32_funcref (func (param nullref f32 f64 i64 v128 v128) (result nullref f32 funcref)))
+ (memory $0 (shared 1 1))
+ (data (i32.const 0) "")
+ (table $0 108 funcref)
+ (global $global$0 (mut i32) (i32.const -1883585))
+ (global $global$1 (mut nullref) (ref.null))
+ (global $global$2 (mut anyref) (ref.null))
+ (global $global$3 (mut nullref) (ref.null))
+ (global $global$4 (mut i32) (i32.const 10))
+ (func $0 (result i32)
+  (i32.const 0)
+ )
+ (func $1 (param $0 i64) (result nullref)
+  (ref.null)
+ )
+ (func $2
+  (nop)
+ )
```

Comment:
There are an awful lot of junk functions in here that are nothing but nops or unreachable. My guess is that the problem will still reproduce if we remove them, and that would make the test much clearer. Even then, I'd almost rather not have a test than check in this beast 😬 @kripken wdyt?

Reply:
Let me try to reduce it by hand :)

Reply:
Managed to isolate the problematic expression tree. Also tried to reduce that tree, but the first two attempts made the error go away, so I guess it's best to preserve its full beauty? :)